ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ConfigManagementServiceIT.scala

```diff
@@ -177,4 +186,89 @@ final class ConfigManagementServiceIT extends LedgerTestSuite {
+    // Multiple config changes should never cause duplicates. Participant adds extra entropy to the
+    // submission id to ensure client does not inadvertently cause problems by poor selection
+    // of submission ids.
+    for {
+      (req1, req2) <- generateRequest(alpha).map(r =>
+        r -> r
+          .update(_.configurationGeneration := r.configurationGeneration + 1)
+      )
+      _ <- ignoreNotAuthorized(alpha.setTimeModel(req1))
+      _ <- ignoreNotAuthorized(beta.setTimeModel(req2))
+    } yield ()
+
+  })
+
+  test(
+    "DuplicateSubmissionIdWhenSubmissionsIncorrect",
+    "Duplicate submission ids are accepted when config changes are invalid",
+    allocate(NoParties, NoParties),
+    runConcurrently = false,
+  )(implicit ec => { case Participants(Participant(alpha), Participant(beta)) =>
+    for {
+      (good, bad) <- generateRequest(alpha).map(r =>
+        r ->
+          r.update(_.configurationGeneration := r.configurationGeneration + 2)
+      )
+      failure1 <- beta.setTimeModel(bad).mustFail(mismatchingGenerations)
+      _ <- ignoreNotAuthorized(alpha.setTimeModel(good))
+      failure2 <- beta.setTimeModel(bad).mustFail(mismatchingGenerations)
+    } yield {
+      assertGenerationMispatchOrNotAuthorized(failure1)
+      assertGenerationMispatchOrNotAuthorized(failure2)
+    }
+  })
+
+  def generateRequest(
+      participant: ParticipantTestContext
+  )(implicit ec: ExecutionContext): Future[SetTimeModelRequest] =
+    for {
+      response <- participant.getTimeModel()
+      oldTimeModel = response.timeModel.getOrElse(
+        throw new AssertionError("Expected time model to be defined")
+      )
+      t1 <- participant.time()
+      req = participant.setTimeModelRequest(
+        mrt = t1.plusSeconds(30),
+        generation = response.configurationGeneration,
+        newTimeModel = oldTimeModel,
+      )
+    } yield req
+
+  private val notAuthorizedPattern = "not authorized"
+  private val mismatchingGenerations = "Mismatching configuration generation"
+
+  // On some ledger implementations only one participant is allowed to modify config, others will
+  // fail with an authorization failure.
+  def ignoreNotAuthorized(
+      call: Future[SetTimeModelResponse]
+  )(implicit executionContext: ExecutionContext): Future[Option[SetTimeModelResponse]] =
+    call.transform {
+      case Success(value: SetTimeModelResponse) =>
+        Success(Some(value))
+      case Failure(GrpcException(GrpcStatus(Status.Code.ABORTED, Some(msg)), _))
+          if (notAuthorizedPattern.r.pattern.matcher(msg).find()) =>
+        Success(None)
+      case Failure(failure) =>
+        Failure(failure)
+
+    }
+
+  def assertGenerationMispatchOrNotAuthorized(failure: Throwable): Unit =
+    failure match {
+      case GrpcException(GrpcStatus(Status.Code.ABORTED, Some(msg)), _)
+          if (notAuthorizedPattern.r.pattern.matcher(msg).find()) =>
```

Comment:
This relies on the error description. This is ok if we know the error comes from the participant, but we have no control over the description if it comes from the ledger (see previous comment).